core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -627,8 +628,38 @@ class BrokerServer(
   }
 
   private def createShareCoordinator(): ShareCoordinator = {
-    val shareConfig = new ShareCoordinatorConfig(config.shareCoordinatorStateTopicSegmentBytes)
-    new ShareCoordinatorService(shareConfig)
+    val time = Time.SYSTEM
```

Comment:
Query: why don't we check `config.isShareGroupEnabled` here before creting the instance?

Reply:
we could do that